### ISSUE TITLE
How multi_gpu_model training works without gradient averager ?

### ISSUE THREAD

**Issue by iperov** [keras-team/keras#11976]
Look at official tensorflow multi gpu training example https://github.com/tensorflow/tensorflow/blob/r0.7/tensorflow/models/image/cifar10/cifar10_multi_gpu_train.py
it has gradient averager and loss applied in each GPU scope.

But keras version only merges output batches.

**Comment by msymp** [comment]
@iperov , can you please provide the full code and system environment you are currently using? Thanks.

**Comment by iperov** [comment]
@msymp  ?? srsly ? why you need it.

**Comment by iperov** [comment]
@msymp just look at keras multi_gpu_model
https://github.com/keras-team/keras/blob/d059890d0342955e968fdf97b5a90d19c9d68b4e/keras/utils/multi_gpu_utils.py

can you show me where there are gradient averager and loss per gpu in this func?

**Comment by jvishnuvardhan** [comment]
@iperov Sorry for the delay in response. Is this still an issue with recent version TF and Keras. Could you please fill the issue template [here](https://github.com/keras-team/keras/issues/new/choose). Could you update them if they are relevant in your case, or leave them as N/A? Along with the template, please provide as many details as possible to find the root cause of the issue. Thanks!

**Comment by iperov** [comment]
@jvishnuvardhan yes it is still an issue.
I cannot see gradient averager and per-gpu-loss https://github.com/keras-team/keras/blob/master/keras/utils/multi_gpu_utils.py
so multi_gpu_model does not work for multi-gpu training
